validation/hooks_stdin.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	tap "github.com/mndrix/tap-go"
+	rspecs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-tools/specerror"
+	"github.com/opencontainers/runtime-tools/validation/util"
+	uuid "github.com/satori/go.uuid"
+)
+
+func stdinStateCheck(outputDir, hookName string, expectedState rspecs.State) error {
+	var state rspecs.State
+	data, err := ioutil.ReadFile(filepath.Join(outputDir, hookName))
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(data, &state)
+	if err != nil {
+		return err
+	}
+
+	if state.ID != expectedState.ID {
+		return fmt.Errorf("wrong container ID %q in the stdin of %s hook, expected %q", state.ID, hookName, expectedState.ID)
+	}
+
+	if state.Bundle != expectedState.Bundle {
+		return fmt.Errorf("wrong bundle directory %q in the stdin of %s hook, expected %q", state.Bundle, hookName, expectedState.Bundle)
+	}
+
+	if hookName != "poststop" && state.Pid != expectedState.Pid {
+		return fmt.Errorf("wrong container process ID %q in the stdin of %s hook, expected %q", state.Version, hookName, expectedState.Version)
+	}
+
+	if !reflect.DeepEqual(state.Annotations, expectedState.Annotations) {
+		return fmt.Errorf("wrong annotations \"%v\" in the stdin of %s hook, expected \"%v\"", state.Annotations, hookName, state.Annotations)
```

Comment:
The last parameter should be `expectedState.Annotations`.

Reply:
updated.